### PR TITLE
Support the "C++ Language Standard" project setting

### LINKED ISSUE #297: JUCE 5.2.0 C++ standard version
Hi @McMartin,

I post this issue to report you a bug with the last version of FRUT (the one on master) and JUCE 5.2.0. I've tried to integrate FRUT to a project that is using some c++14 features, and the compilation failed when reaching the c++14 code.
I've investigated a little and found that, in the CMaleLists.txt file generated by FRUT, I have the line `CXX_LANGUAGE_STANDARD "Use Default"` for the export `Xcode (MacOSX)` in Debug and Release mode. On the other project which are using only C++ 11 features, the projects compiles, but the same line `CXX_LANGUAGE_STANDARD "Use Default"` is present.

I've looked a bit further and found that in the jucer files of those projects, the C++ standard version is specified like that : `cppLanguageStandard="11"` for C++11 and cppLanguageStandard="14" for C++14, while in the FRUT Reprojucer main.cpp file, the value isn't the same ([link to the part of the source code here](https://github.com/McMartin/FRUT/blob/master/Jucer2Reprojucer/main.cpp#L1118))

I hope this will help you to have a better support of JUCE 5.2.0 ;)

PS : For the fun, I've tried the different option for the field `cppLanguageStandard` in the Projucer  of JUCE 5.2.0, there are 3 options : "C++11", "C++14" and "Use Latest". Each of them are specified in the projucer respectively by `cppLanguageStandard="11"`, `cppLanguageStandard="14"` and `cppLanguageStandard="latest"`

### PR DESCRIPTION
**Pull request by McMartin**
Resolves #297 

@lethal-guitar, @MartyLake: please review, thanks!

### CODE REVIEW

**Comment by lethal-guitar** [comment]
Gonna start looking at this one as well, might take some time before I can finish though.

**Comment by McMartin** [comment]
@lethal-guitar: no problem at all, take the time you need. Thank you very much for reviewing!

File: cmake/Reprojucer.cmake [hunk 1]
```diff
@@ -2604,13 +2604,9 @@ function(__set_cxx_language_standard_properties target)
       endif()
       if(cxx_language_standard MATCHES "^GNU\\+\\+")
         set_target_properties(${target} PROPERTIES CXX_EXTENSIONS ON)
-      elseif(cxx_language_standard MATCHES "^C\\+\\+")
```
[the remark this answers is not in the document]
Comment (McMartin) [reply cmake/Reprojucer.cmake]: Sounds like a very good idea! I'll do it once you are done with the review.

File: cmake/Reprojucer.cmake [hunk 2]
```diff
@@ -2577,10 +2577,10 @@ endfunction()
 
 function(__set_cxx_language_standard_properties target)
 
-  if(APPLE)
-    set_target_properties(${target} PROPERTIES CXX_EXTENSIONS OFF)
-    set_target_properties(${target} PROPERTIES CXX_STANDARD 11)
+  set_target_properties(${target} PROPERTIES CXX_EXTENSIONS OFF)
```
[the remark this answers is not in the document]
Comment (McMartin) [reply cmake/Reprojucer.cmake]: I like this, thanks!

File: cmake/Reprojucer.cmake [hunk 3]
```diff
@@ -2580,54 +2580,54 @@ function(__set_cxx_language_standard_properties target)
...
+    elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")
+      if(DEFINED JUCER_CXX_STANDARD_TO_USE)
+        if(JUCER_CXX_STANDARD_TO_USE MATCHES "03$")
+          set_target_properties(${target} PROPERTIES CXX_STANDARD 98)
+        elseif(JUCER_CXX_STANDARD_TO_USE MATCHES "14$")
+          set_target_properties(${target} PROPERTIES CXX_STANDARD 14)
+        endif()
       endif()
     endif()
-  endif()
```
[the remark this answers is not in the document]
Comment (McMartin) [reply cmake/Reprojucer.cmake]: This is an artifact of the diff of https://github.com/McMartin/FRUT/pull/300/commits/50a48d7160f2da6785cb5f48d7d8b1bc097eeb6f. Put `?w=1` at the end of the URL to see the diff with space changes ignored.

[the remark this answers is not in the document]
Comment (McMartin) [reply cmake/Reprojucer.cmake]: And sure the `endif` comes back in the next commit, we are surrounding the whole block with `if/else/endif`.

File: cmake/Reprojucer.cmake [hunk 4]
```diff
@@ -2632,6 +2575,72 @@ function(__set_common_target_properties target)
...
+      endif()
+      if(cxx_language_standard MATCHES "98$")
+        set_target_properties(${target} PROPERTIES CXX_STANDARD 98)
+      elseif(cxx_language_standard MATCHES "11$")
+        set_target_properties(${target} PROPERTIES CXX_STANDARD 11)
+      elseif(cxx_language_standard MATCHES "14$")
+        set_target_properties(${target} PROPERTIES CXX_STANDARD 14)
+      endif()
+    endif()
+  elseif(MSVC)
```
[the remark this answers is not in the document]
Comment (McMartin) [reply cmake/Reprojucer.cmake]: That's a good idea! Addressed in 719f375 (which won't apply cleanly to 5bc1075, but I'll deal with it when rebasing).

**Comment by McMartin** [comment]
TODO when rebasing:
- squash `Remove redundant code` into `Extract setting defaults`
- reword `Extract setting defaults` to `Unconditionally set defaults`

**Comment by McMartin** [comment]
I close this PR because there are other changes that should be done before and I don't want @lethal-guitar or @MartyLake to waste time reviewing this now.

**Comment by McMartin** [comment]
@lethal-guitar if you want to continue your review, you can now do so in https://github.com/McMartin/FRUT/pull/318.

**Pull request closed without merging**